confmap/provider/eecprovider/provider.go

```diff
@@ -0,0 +1,199 @@
+// Copyright The OpenTelemetry Authors
+// Copyright Dynatrace LLC
+// SPDX-License-Identifier: Apache-2.0
+
+// This file was copied from the configurablehttpprovider package and modified
+// to support configuration options through query parameter-encoded URL fragments.
+// Additionally, support for two schemes (HTTP/HTTPS) was dropped in favor of
+// configuring TLS through an option on the provider.
+
+package eecprovider // import "github.com/Dynatrace/dynatrace-otel-collector/confmap/provider/eecprovider"
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type SchemeType string
+
+const (
+	EECScheme SchemeType = "eec"
+)
+
+const (
+	RefreshInterval = "refresh-interval"
+	AuthHeader      = "auth-header"
+	AuthFile        = "auth-file"
+	AuthEnv         = "auth-env"
+	Insecure        = "insecure"
+)
+
+type provider struct {
+	caCertPath         string // Used for tests
+	insecureSkipVerify bool   // Used for tests
+	ctx                context.Context
+	cancel             context.CancelFunc
+}
+
+var _ confmap.Provider = (*provider)(nil)
+
+// NewFactory returns a factory for a confmap.Provider that reads the configuration from an https server.
+//
+// This Provider supports "eec" scheme.
+//
+// One example for an HTTPS URI is eec://localhost:3333/getConfig
+func NewFactory() confmap.ProviderFactory {
+	return confmap.NewProviderFactory(newProvider)
+}
+
+func newProvider(_ confmap.ProviderSettings) confmap.Provider {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &provider{ctx: ctx, cancel: cancel}
+}
+
+// Create the client based on the type of scheme that was selected.
+func (p *provider) createClient(insecure bool) (*http.Client, error) {
+	if insecure {
+		return &http.Client{}, nil
+	}
+
+	pool, err := x509.SystemCertPool()
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to create a cert pool: %w", err)
+	}
+
+	if p.caCertPath != "" {
+		cert, err := os.ReadFile(filepath.Clean(p.caCertPath))
+
+		if err != nil {
+			return nil, fmt.Errorf("unable to read CA from %q URI: %w", p.caCertPath, err)
+		}
+
+		if ok := pool.AppendCertsFromPEM(cert); !ok {
+			return nil, fmt.Errorf("unable to add CA from uri: %s into the cert pool", p.caCertPath)
+		}
+	}
```

Comment:
This is all only required for testing right?

Reply:
Right. there's no way to set `caCertPath` outside of this package.